File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
    
        String[] str=new String[numbers.length];
        for(int i = 0;i<str.length;i++){
            str[i]=String.valueOf(numbers[i]);
        }
        
        Arrays.sort(str,(s1,s2)->(s2+s1).compareTo(s1+s2));
        
        for(String s:str){
            answer+=s;
        }
        
        if(answer.charAt(0) == '0'){
            answer = "0";
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
import functools

def compare(x,y):
    if x+y>y+x: return 1
    elif x+y==y+x: return 0
    else: return -1

def solution(numbers):
    numbers=list(map(str,numbers))
    numbers.sort(key=functools.cmp_to_key(compare),reverse=True )
    return str(int(''.join(numbers)))